### PR TITLE
refactor(messages): nested messages inside of channels

### PR DESCRIPTION
**Pull request by LukeBrandon**
### Description of changes:
- Moved all `/messages` urls to `/channels/:channel_id/messages` and getting `channel_id`from the path
- Updated queries for `messages` to filter by `channel_id`

### CODE REVIEW

**Comment by FlynnOConnell** [comment]
@LukeBrandon unfortunately, you didn't run `cargo sqlx prepare` before this commit. We need that command to enable building in sqlx offline mode using the query! macros. 

[See here for more information!](https://github.com/launchbadge/sqlx/blob/main/sqlx-cli/README.md#enable-building-in-offline-mode-with-query)

**Review by FlynnOConnell — COMMENTED** [review]
few things

**Comment by LukeBrandon** [comment]
I need to fix the routing issue that is happening with conflicting urls

File: backend/src/controllers/message.rs [hunk 1]
```diff
@@ -84,20 +87,24 @@ pub async fn create_message(
     Path(channel_id): Path<i64>,
     Extension(pool): Extension<PgPool>,
     Json(message): Json<CreateMessageSchema>,
-) -> Result<(StatusCode, Json<MessageModel>), CustomError> {
+    ) -> Result<(StatusCode, Json<MessageModel>), CustomError> {
     if message.message.is_empty() {
-        return Err(CustomError::BadRequest);
+        return Err(CustomError::InternalServerError);
```
Comment (LukeBrandon) [inline backend/src/controllers/message.rs]: change this back

**Comment by FlynnOConnell** [comment]
looks swell

**Review by FlynnOConnell — APPROVED** [review]

**Pull request merged**